In [1]:
import pandas as pd

e = pd.read_csv("effect_size_by_group.csv")

summary = (e.groupby(["source_dataset", "family_history_flag"])
             .agg(mean_RD=("RD_high_low", "mean"),
                  median_RD=("RD_high_low", "median"),
                  mean_RR=("RR_high_low", "mean"),
                  count=("RD_high_low", "count"))
             .reset_index())

print(summary)

  source_dataset  family_history_flag   mean_RD  median_RD   mean_RR  count
0   professional                    0  0.037200   0.037195  4.285714      6
1   professional                    1  0.058903   0.069767  1.363636      5
2        student                    0  0.345630   0.350044  2.006616      6
3        student                    1  0.362434   0.374505  2.000389      6


In the student dataset, the average risk difference (High–Low financial bucket) is ~0.35, whereas in the professional dataset it is only ~0.04–0.06, suggesting a much stronger association between financial strain and depression among students. Risk ratios fluctuate more in the professional sample due to small baseline (Low) rates, so RD is used as the more stable comparison.